src/index.js

```diff
@@ -15,14 +15,22 @@ export function getPluginsThatDontSatisfyModuleRange (plugins, range) {
   return _.pickBy(plugins, ({ node }) => !semver.satisfies(normalizeSemver(node), range))
 }
 
-export function getModuleNeedsBabel (pkg, { hasEsNextInMainFields, hostEngines } = {}) {
-  const { engines } = pkg
+export function getModuleNeedsBabel (pkg, { hasEsNextInMainFields, hostEngines, defaultEngines } = {}) {
+  let { engines } = pkg
   const hasEsNextField = getHasESNextField(Object.keys(pkg))
   // always transpile if we have an esnext field
   if (hasEsNextInMainFields && hasEsNextField) {
     return true
   }
-  if (!engines) return false
+  if (!engines) {
+    if (defaultEngines) {
+      if (typeof defaultEngines !== 'object') {
+        return true
+      }
+      engines = defaultEngines
+    }
+    return false
```

Comment:
If we `return false` here, we will never be able to use `engines` assigned as `defaultEngines`. Should we remove this line?

```suggestion
```

Reply:
Oops, it's supposed to be an 'else' to handle defaultEngines = false.